torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -511,6 +511,125 @@ std::vector<int64_t> ComputeShapeFromReshape(
   return final_shape;
 }
 
+// The logic here is similar to its overloading function above.
+c10::optional<::c10::SymbolicShape> ComputeShapeFromReshape(
+    Node* n,
+    const c10::SymbolicShape& input_shape,
+    const c10::SymbolicShape& reshape,
+    int opset_version) {
+  std::vector<c10::ShapeSymbol> input_shape_vector =
+      input_shape.sizes().value();
+  std::vector<c10::ShapeSymbol> reshape_vector = reshape.sizes().value();
+  TORCH_INTERNAL_ASSERT(
+      input_shape_vector.size() > 0 || reshape_vector.size() > 0,
+      "Reshape node should have at least one input size > 0 when constant folding.");
+  if (reshape_vector.size() == 0) {
+    return input_shape;
+  }
+  if (input_shape_vector.size() == 0) {
+    return reshape;
+  }
+
+  auto reshape_size = static_cast<int>(reshape_vector.size());
+  auto is_zero = [](c10::ShapeSymbol& ss) { return ss.value() == 0; };
+  auto it_0 =
+      std::find_if(reshape_vector.begin(), reshape_vector.end(), is_zero);
+  auto reshape_has_zero = it_0 != reshape_vector.end();
+
+  int allowzero = 0;
+  if (opset_version >= 14 && n->hasAttributeS("allowzero")) {
+    allowzero = n->i(attr::allowzero);
+    if (allowzero == 1 && reshape_has_zero) {
+      return reshape;
```

Comment:
If `reshape_has_zero`, we cannot use the below algorithm to calculate what -1 is, because it will divide by zero. So when `reshape_has_zero`, just return `reshape`. 

If `allowzero` is set, we have regular handling for -1 when `reshape_has_zero=False`. But when `reshape_has_zero=True`, we cannot divide by -1 so we return `reshape` here. 

Let me add a TODO here if we want to investigate -1 case when `if (allowzero == 1 && reshape_has_zero)`